packages/react-native/ReactCommon/react/renderer/graphics/platform/ios/react/renderer/graphics/HostPlatformColor.h

```diff
@@ -8,32 +8,108 @@
 #pragma once
 
 #include <react/renderer/graphics/ColorComponents.h>
+#include <react/utils/hash_combine.h>
 #include <cmath>
 
 namespace facebook::react {
 
-using Color = int32_t;
+enum ColorKind { UndefinedKind, ComponentsKind, DynamicKind };
+
+struct DynamicColor {
+  int32_t lightColor = 0;
+  int32_t darkColor = 0;
+  int32_t highContrastLightColor = 0;
+  int32_t highContrastDarkColor = 0;
+
+  bool operator==(const DynamicColor& other) const {
+    return lightColor == other.lightColor && darkColor == other.darkColor &&
+        highContrastLightColor == other.highContrastLightColor &&
+        highContrastDarkColor == other.highContrastDarkColor;
+  }
+};
+
+class Color {
+ public:
+  Color() : Color(UndefinedKind) {}
+  Color(DynamicColor dynamicColor)
+      : kind_(DynamicKind),
+        dynamicColor_(dynamicColor),
+        color_(dynamicColor.lightColor) {}
+  Color(ColorComponents components) {
+    float ratio = 255;
+    kind_ = ComponentsKind;
+    color_ = ((int)round(components.alpha * ratio) & 0xff) << 24 |
+        ((int)round(components.red * ratio) & 0xff) << 16 |
+        ((int)round(components.green * ratio) & 0xff) << 8 |
+        ((int)round(components.blue * ratio) & 0xff);
+  }
```

Comment:
This is breaking.
we can't change the definition of Color, as it will break all the code path that expec color to be an `int32_t`. This is "easy" within the framework, but there could be 3rd party libraries that uses this type, and this change will break them.

I think C++ has some construct that let you make this syntax valid
```c++
Color c = 0xAA44FF; 
```
I think that they are [converting constructors](https://en.cppreference.com/w/cpp/language/converting_constructor), IIRC. So we should be able to make it non breaking when a simple integer is passed.

To consume it, I think that there is a similar concept, using the [implicit conversion operator](https://en.cppreference.com/w/cpp/language/cast_operator). 

With these updates, the change will be not breaking.